core/vm/interface.go

```diff
@@ -26,58 +26,123 @@ import (
 
 // StateDB is an EVM database for full state querying.
 type StateDB interface {
+	// CreateAccount creates a new account
+	//
+	// If an account already exists at the address, it should transfer the balance of the old
+	// address to the new one.
```

Comment:
This describes the behaviour of the implementation, not the semantics of the statedb. From statedb perspective "if the account already exists at the address, this operation is moot". 
I think?

Reply:
Thanks, I was not sure if that's an implementation detail or expected behaviour, but dropped it at [b1fd558](https://github.com/ethereum/go-ethereum/pull/28731/commits/b1fd5588c3ddf5f4a1e5b5e53c246e02f9ba537b)